### PR TITLE
Update datastatus to accept enum

### PR DESCRIPTION
**Pull request by fredms**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_set_operations.py [hunk 1]
```diff
@@ -217,6 +218,8 @@ def begin_backfill(
         :paramtype compute_resource: ~azure.ai.ml.entities.MaterializationComputeResource
         :keyword spark_configuration: Specifies the spark compute settings.
         :paramtype spark_configuration: dict[str, str]
+        :keyword data_status: Specified the data status that you want to backfill.
+        :paramtype data_status: list[str or ~azure.mgmt.machinelearningservices.models.DataAvailabilityStatus]
```
Comment (shail2208) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_set_operations.py:222]: Is this an enum from the rest client?

Comment (fredms) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_set_operations.py:222]: yes, it's from rest client.

**Review by qjxu — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_set_operations.py [hunk 2]
```diff
@@ -217,6 +218,8 @@ def begin_backfill(
         :paramtype compute_resource: ~azure.ai.ml.entities.MaterializationComputeResource
         :keyword spark_configuration: Specifies the spark compute settings.
         :paramtype spark_configuration: dict[str, str]
+        :keyword data_status: Specified the data status that you want to backfill.
```
Comment (bastrik) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_set_operations.py:221]: Nit: "Specifies the data status of the backfill job"

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_set_operations.py (hunk 1 above)
Comment (fredms) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_set_operations.py:223]: updated

**Review by shail2208 — APPROVED** [review]

**Pull request merged**